kmip/services/server/engine.py

```diff
@@ -0,0 +1,404 @@
+# Copyright (c) 2016 The Johns Hopkins University/Applied Physics Laboratory
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import logging
+import threading
+import time
+
+import kmip
+
+from kmip.core import enums
+from kmip.core import exceptions
+
+from kmip.core.messages import contents
+from kmip.core.messages import messages
+
+from kmip.core.messages.payloads import discover_versions
+from kmip.core.messages.payloads import query
+
+from kmip.core import misc
+
+from kmip.services.server.crypto import engine
+
+
+class KmipEngine(object):
+    """
+    A KMIP request processor that acts as the core of the KmipServer.
+
+    The KmipEngine contains the core application logic for the KmipServer.
+    It processes all KMIP requests and maintains consisten state across
+    client connections.
+
+    Features that are not supported:
+        * KMIP versions > 1.1
```

Comment:
I guess my thinking falls back to the features that distinguish the different versions. 1.2 adds a lot of new operations that we don't support and, in my mind, I don't want to claim 1.2 support if the bulk of that implies a lot of functionality that would be missing. Syntactically, you're right, we should be compatible. I just don't want to claim more than we should.